packages/next/build/webpack/loaders/next-flight-server-loader.ts

```diff
@@ -58,29 +67,49 @@ async function parseImportsInfo(
       case 'ImportDeclaration': {
         const importSource = node.source.value
         if (!isClientCompilation) {
+          // Server compilation for .server.js.
+          if (isServerComponent(importSource)) {
+            continue
+          }
+
+          const importDeclarations = source.substring(
+            lastIndex,
+            node.source.span.start - beginPos
+          )
+
           if (
             !(
-              isClientComponent(importSource, pageExtensions) ||
+              isClientComponent(importSource) ||
               isNextComponent(importSource) ||
               isImageImport(importSource)
             )
           ) {
-            continue
+            if (
+              ['react/jsx-runtime', 'react/jsx-dev-runtime'].includes(
```

Comment:
This looks tricky, is there any other library having the similar reason that we need to skip?

Reply:
Let's add a comment for it later

Reply:
This is a special case to avoid the [Duplicate React](https://reactjs.org/warnings/invalid-hook-call-warning.html#duplicate-react) error. Since we already include React in the SSR runtime, here we can't create a new module with the `?__rsc_server` query.

Reply:
We might be able to change that with a webpack alias (`react/jsx-runtime?__rsc_server__` -> `react/jsx-runtime`), but not sure if that's more readable than this.